js/map.js

```diff
@@ -0,0 +1,117 @@
+(function () {
+  'use strict';
+
+  var nearbyListingsTemplate = {
+    author: {
+      avatar: ""
+    },
+    offer: {
+      title: "",
+      address: `{{location.x}}, {{location.y}}`,
+      price: 0,
+      type: "",
+      rooms: 0,
+      guests: 0,
+      checkin: "",
+      checkout: "",
+      features: [],
+      description: "",
+      photos: [],
+    },
+    location: {
+      x: 1,
+      y: 130
+    }
+  };
+
+  var titleArray = ['Big cosy flat',
+    'Small uncosy flat',
+    'Huge beautiful palace',
+    'Small horrible palace',
+    'Beautiful guest house',
+    'Ugly inhospitable house',
+    'Cosy bungalow far from the sea',
+    'Uncosy bungalow in water up to your knees'
+  ];
+
+  var propertyTypeArray = ['palace', 'flat', 'house', 'bungalow'];
+  var checkinTimeArray = ['12:00', '13:00', '14:00'];
+  var checkoutTimeArray = ['12:00', '13:00', '14:00'];
+  var featuresArray = ['wifi', 'dishwasher', 'parking', 'washer', 'elevator', 'conditioner'];
+  var photosArray = ['http://o0.github.io/assets/images/tokyo/hotel1.jpg',
+    'http://o0.github.io/assets/images/tokyo/hotel2.jpg',
+    'http://o0.github .io/assets/images/tokyo/hotel3.jpg'
+  ];
+
+  function generateRandomNumbersArray(arraySize, maxNumber, shiftNumber) {
+    if (arraySize < 1 || maxNumber < 1) {
+      return;
+    }
+
+    var randomNumberArray = [];
+
+    while (randomNumberArray.length < arraySize) {
+      var randomNumber = Math.floor(Math.random() * Math.floor(maxNumber)) + shiftNumber || 0;
+      if (randomNumberArray.indexOf(randomNumber) > -1) {
+        continue;
+      }
+      randomNumberArray.push(randomNumber);
+    }
+    return randomNumberArray;
+  }
+
+  function getRandomNumberInRange(min, max) {
+
+    min = Math.ceil(min);
+    max = Math.floor(max);
+    return Math.floor(Math.random() * (max - min)) + min;
+  }
+
+
+  function randomListings(number) {
+    var adsArray = [];
+    var randomNum = generateRandomNumbersArray(number, number, 1);
+    var randomFeaturesNumber = getRandomNumberInRange(1, 6);
+    var chooseRandomFeatures = generateRandomNumbersArray(randomFeaturesNumber, featuresArray.length, 0).sort(function (a, b) {
+      return a - b
+    });
+
+    for (let i = 0; i < number; i++) {
+
+      var randomFeaturesNumber = getRandomNumberInRange(1, 6);
+      var randomPhotosOrder = generateRandomNumbersArray(3, photosArray.length, 0);
+
+      var chooseRandomFeatures = generateRandomNumbersArray(randomFeaturesNumber, featuresArray.length, 0).sort(function (a, b) {
+        return a - b
+      });
+
+      var x = getRandomNumberInRange(0, 630);
+      var y = getRandomNumberInRange(130, 630);
+
+
+      adsArray.push(JSON.parse(JSON.stringify(nearbyListingsTemplate)));
```

Comment:
Think about it in the following way: 
1) You create an empty array, which you did
2) Then you call a function that generates you a random listing
3) You add the returned generated object to the array

you don't mutate already created and pushed to the array object, it's not efficient. 
Try refactoring this and it might change how it looks here.